### PR TITLE
chore(deps): rm go-jose Cxb6dee8d5-b814 high vuln

### PR DESCRIPTION
**Pull request by fengshunli**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

see https://devhub.checkmarx.com/cve-details/Cxb6dee8d5-b814/

### CODE REVIEW

**Comment by blakepettersson** [comment]
The tests are failing, probably due to the fact that the go-jose implementation does a bunch of magic when it comes to marshaling the key (see https://github.com/go-jose/go-jose/blob/833bb274d109632dc533465252426d82f79c226e/jwk.go#L105-L125). IMO the way to go would be to instead change the dependency to `go-jose/go-jose/v3`.

File: util/test/testutil.go [hunk 1]
```diff
@@ -107,6 +108,30 @@ rawN1NFkKr5MCxl5Q4veocU94ThOlFdYgreyVX6s40ZL1eF0RvAQ+e0oFT7SfCHu
...
+
+	// X.509 certificate chain, parsed from `x5c` header.
+	Certificates []*x509.Certificate
+	// X.509 certificate URL, parsed from `x5u` header.
+	CertificatesURL *url.URL
+	// X.509 certificate thumbprint (SHA-1), parsed from `x5t` header.
+	CertificateThumbprintSHA1 []byte
+	// X.509 certificate thumbprint (SHA-256), parsed from `x5t#S256` header.
+	CertificateThumbprintSHA256 []byte
+}
```
Comment (crenshaw-dev) [inline util/test/testutil.go:133]: Are these just copy/pasted from go-jose? If so, wanna add a comment linking to the original source and explaining why we copied them?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: util/test/testutil.go (hunk 1 above)
Comment (crenshaw-dev) [reply util/test/testutil.go:133]: Just to stay 100% in the clear on the Apache license, I'd also drop the code comments or replace them with your own if those are copy/pasted.

Comment (fengshunli) [reply util/test/testutil.go]: This package go-jose/go-jose/v3 is not used much. You need to replace v3 or copy the structure.

**Review by blakepettersson — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**